### DISCUSSION TITLE
BepInEx Mono cannot build with error "The type 'Object' is defined in an assembly that is not referenced"

### DISCUSSION THREAD

**Discussion by datvm** [BepInEx/BepInEx#922] [Help and Q&A] [answered]
I tried following [this doc](https://docs.bepinex.dev/master/articles/dev_guide/plugin_tutorial/2_plugin_start.html?tabs=tabid-unitymono) to create an example plugin. I could run `dotnet restore` but not `dotnet build` due to this error:

```
Error (active)	CS0012	The type 'Object' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.1.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'.	TestPlugin	D:\Personal\Mods\PlanetCrafter\TestPlugin\Plugin.cs	7	
```

Any idea what is happening? I have .NET 8 SDK installed (my day-to-day work) with .NET 4.6 *Targeting pack* as well because I think the game I am modding for is using .NET 4.6.

I have even tried adding this but it doesn't affect the build in any way.

```
<Reference Include="netstandard" />
```

**Comment by Oshroth** [comment, marked as answer]
You are targeting the wrong framework for the game. Some Unity games use `netstandard2.0` or `netstandard2.1` not `net4.6`.
In your case, PlanetCrafter uses `netstandard2.1`
> If the game has netstandard.dll in <Game Name>_Data/Managed folder, your TFM is netstandard2.0

For Planet Crafter, you need to use `dotnet new bepinex5plugin -n <Plugin Name> -T netstandard2.1 -U 2023.2.19`

**Reply by datvm** [reply]
Thank you. I remember I ended up copying the setup from another mod but thanks :)

**Reply by Oshroth** [reply]
I suspected you might have solved the problem but this was the top Google search result for this error and none of the other results had a fix, so I added the solution in case anyone else needed it.
